packages/interaction-tracking/src/InteractionTracking.js

```diff
@@ -0,0 +1,341 @@
+      return callback();
+    } finally {
+      ((interactionsRef: any): InteractionsRef).current = prevInteractions;
+    }
+  }
+}
+
+export function unsubscribe(subscriber: Subscriber): void {
+  if (enableInteractionTracking && enableInteractionTrackingObserver) {
+    ((subscriberRef: any): SubscriberRef).current = null;
+  }
+}
+
+export function wrap(
+  callback: Function,
+  threadID: number = DEFAULT_THREAD_ID,
+): Function {
+  if (!enableInteractionTracking) {
+    return callback;
+  }
+
+  const wrappedInteractions = ((interactionsRef: any): InteractionsRef).current;
+
+  if (enableInteractionTrackingObserver) {
+    const subscriber = ((subscriberRef: any): SubscriberRef).current;
+    if (subscriber !== null) {
+      subscriber.onWorkScheduled(wrappedInteractions, threadID);
+    }
+
+    // Update the pending async work count for the current interactions.
+    // Update after calling subscribers in case of error.
+    wrappedInteractions.forEach(interaction => {
+      interaction.__count = ((interaction.__count: any): number) + 1;
+    });
+  }
+
+  const wrapped = (...args) => {
+    const prevInteractions = ((interactionsRef: any): InteractionsRef).current;
+    ((interactionsRef: any): InteractionsRef).current = wrappedInteractions;
+
+    if (enableInteractionTrackingObserver) {
+      const subscriber = ((subscriberRef: any): SubscriberRef).current;
+
+      try {
+        let caughtError;
+        let didCatch = false;
+        let returnValue;
+
+        try {
+          if (subscriber !== null) {
+            subscriber.onWorkStarted(wrappedInteractions, threadID);
+          }
+        } catch (error) {
+          didCatch = true;
+          caughtError = caughtError || error;
+        }
+
+        try {
+          returnValue = callback(...args);
+        } catch (error) {
+          didCatch = true;
+          caughtError = caughtError || error;
+        }
+
+        ((interactionsRef: any): InteractionsRef).current = prevInteractions;
+
+        try {
+          if (subscriber !== null) {
+            subscriber.onWorkStopped(wrappedInteractions, threadID);
+          }
+        } catch (error) {
+          didCatch = true;
+          caughtError = caughtError || error;
+        }
+
+        if (didCatch) {
+          throw caughtError;
+        } else {
+          return returnValue;
+        }
+      } finally {
+        // Update pending async counts for all wrapped interactions.
+        // If this was the last scheduled async work for any of them,
+        // Mark them as completed.
+        wrappedInteractions.forEach(interaction => {
+          interaction.__count = ((interaction.__count: any): number) - 1;
+
+          if (
+            subscriber !== null &&
+            ((interaction.__count: any): number) === 0
+          ) {
+            subscriber.onInteractionScheduledWorkCompleted(interaction);
+          }
+        });
+      }
+    } else {
+      try {
+        return callback(...args);
```

Comment:
Nit: I think this is fine and Babel will compile it to `callback.apply(undefined, arguments)` but to be safe I usually prefer to manually use `apply`.